test/functional/legacy/074_global_var_in_viminfo_spec.lua

```diff
@@ -0,0 +1,70 @@
+-- Tests for storing global variables in the .viminfo file
+
+-- TODO: We need a way to run this test with a real (dummy?) viminfo file.  It
+-- will not work with the current session which executes "nvim ... -i NONE
+-- ..." as this inhibits any kind of viminfo writing.  See ":h -i".
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+local os = require('os')
+
+describe('storing global variables in viminfo files', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      eof]=])
+
+    -- TODO: Is 'visualbell' meaningfull in the new test environment?
+    -- This will cause a few errors, do it silently.
+    source([[
+      set visualbell
+      set viminfo+=!
+      let MY_GLOBAL_DICT={'foo': 1, 'bar': 0, 'longvarible': 1000}
+    ]])
+    -- Store a really long list, so line wrapping will occur in viminfo file.
+    source([[
+      let MY_GLOBAL_LIST=range(1,100)
+      wv! Xviminfo
+      unlet MY_GLOBAL_DICT
+      unlet MY_GLOBAL_LIST
+    ]])
+    -- If the (still failing) rv! command is included in the above source(),
+    -- the test errors due to output on stderr.
+    execute('rv! Xviminfo')
+    -- TODO: Actually we want to test if the viminfo file was written but the
+    -- test is currently run with "nvim -i NONE ..." which inhibits this.  So
+    -- the test would fail here.
```

Comment:
At the start of this test, you can put
```lua
os.remove("Xviminfo")  -- ensure it does not exist
```
and between the `wv` and `rv` commands, put
```lua
neq(nil, lfs.attributes('Xviminfo'))  -- assert it exists
```
Instead of using `redir > test.out` below (since that's not what's being tested), you can just put
```lua
eq({'foo'=1, 'longvariable'=1000, 'bar'=0}, eval('MY_GLOBAL_DICT'))
```
I could have the syntax a little wrong, but that's the general idea.